### PR TITLE
Added `refFromUrl` and `getDownloadUrl` 

### PR DESCRIPTION
**Pull request by cedvdb**
This has tests

closes https://github.com/atn832/firebase_storage_mocks/issues/18 https://github.com/atn832/firebase_storage_mocks/issues/10

This is basically the code from firestorage repo.

### CODE REVIEW

**Comment by cedvdb** [comment]
Note that it only allows _**valid**_ firebase urls so it could be more lax for easier testing. I'm personally hitting the problem of my mocks url being invalid firebase url when using this, so I'm thinking of changing it to be more lax. 

On the other hand it's more correct like this and it's not hard to make a fake firestorage url

File: lib/src/utils.dart [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/// Returns a path from a given `gs://` URL.
```
Comment (atn832) [inline lib/src/utils.dart:1]: If this comes verbatim from another repository, it is good practice to link it at the beginning of the file. This way if the original repository modifies the logic, we know where to import it from again.

Looks like it came from https://github.com/firebase/flutterfire/blob/790e450e8d6acd2fc50e0232c77a152430c7b3ea/packages/firebase_storage/firebase_storage/lib/src/utils.dart.

File: lib/src/firebase_storage_mocks_base.dart [hunk 2]
```diff
@@ -16,6 +17,26 @@ class MockFirebaseStorage extends Mock implements FirebaseStorage {
     return MockReference(this, path);
   }
 
+  @override
+  Reference refFromURL(String url) {
```
Comment (atn832) [inline lib/src/firebase_storage_mocks_base.dart:22]: Same here. If you copied this from another repo, please include the link. It seems like it came from https://github.com/firebase/flutterfire/blob/3dfc0997050ee4351207c355b2c22b46885f971f/packages/firebase_storage/firebase_storage/lib/src/firebase_storage.dart#L111.

**Review by atn832 — APPROVED** [review]
Thanks for the PR. I'll add the link to the files myself in another PR.

The real code is simple enough that we can probably keep this and not make it more lax.

**Pull request merged**